packages/ui/ViewManager/View.js

```diff
@@ -135,13 +135,19 @@ class View extends React.Component {
 		this.animation = null;
 		this._raf = null;
 		this.state = {
-			entering: !props.appearing
+			entering: !props.appearing,
+			changeDirection: false,
+			reverseTransition: props.reverseTransition
 		};
 	}
 
-	UNSAFE_componentWillReceiveProps (nextProps) {
-		// changeDirection let's us know we need to switch mid-transition
-		this.changeDirection = this.animation ? this.props.reverseTransition !== nextProps.reverseTransition : false;
+	static getDerivedStateFromProps (props, state) {
+		if (state.reverseTransition !== props.reverseTransition) {
```

Comment:
I don't see where `reverseTransition` state is being utilized other than the check here. How does this work?

Reply:
to save it to check with previous prop value later?